lib/bitcoin/BitcoinClient.ts

```diff
@@ -313,6 +325,156 @@ export default class BitcoinClient {
+    const payToScriptAddress = new Address(freezeScriptHash);
+
+    return this.createSpendBitcoreTransactionFromFrozenTransaction(
+      previousFreezeTransaction,
+      previousFreezeUntilBlock,
+      payToScriptAddress);
+  }
+
+  // @ts-ignore
+  private async createSpendToWalletBitcoreTransaction (
+    previousFreezeTransaction: Transaction,
+    previousFreezeUntilBlock: number): Promise<Transaction> {
+
+    return this.createSpendBitcoreTransactionFromFrozenTransaction(
+      previousFreezeTransaction,
+      previousFreezeUntilBlock,
+      this.privateKeyAddress);
+  }
+
+  // @ts-ignore
+  /**
+   * Creates a spend transaction to spend the previously frozen output. The details
+   * on how to create a spend transactions were taken from the BIP65 demo at:
+   * https://github.com/mruddy/bip65-demos/blob/master/freeze.js.
+   *
+   * @param previousFreezeTransaction The previously frozen transaction.
+   * @param previousFreezeUntilBlock The previously frozen transaction's freeze until block.
+   * @param paytoAddress The address where the spend transaction should go to.
+   */
+  private async createSpendBitcoreTransactionFromFrozenTransaction (
+    previousFreezeTransaction: Transaction,
+    previousFreezeUntilBlock: number,
+    paytoAddress: Address): Promise<Transaction> {
+
+    // First create an input from the previous frozen transaction output.
+    const frozenOutputAsInput = this.createBitcoreUnspentOutputFromFrozenTransaction(previousFreezeTransaction, previousFreezeUntilBlock);
+    const previousFreezeAmountInSatoshis = frozenOutputAsInput.satoshis;
+
+    // Now create a spend transaction using the frozen output. Create the transaction with all
+    // inputs and outputs as they are needed to calculate the fee.
+    const spendTransaction = new Transaction()
+                                   .from([frozenOutputAsInput])
+                                   .to(paytoAddress, previousFreezeAmountInSatoshis)
+                                   .lockUntilBlockHeight(previousFreezeUntilBlock);
+
+    const transactionFee = await this.calculateTransactionFee(spendTransaction);
+
+    // We need to set the transaction fee and subtract that fee from the freeze amount.
+    // We cannot just update the existing output (it's readonly), so we need to first remove it,
+    // and add another one with the correct amount.
+    spendTransaction.outputs.shift();
+    spendTransaction.to(paytoAddress, previousFreezeAmountInSatoshis - transactionFee)
+                    .fee(transactionFee);
+
+    // Now update the first input to include it's signature.
+    // Create an input in the format expected for a spend-from-freeze input and set it on the
+    // transaction.
+    const previousFreezeScript = this.createFreezeBitcoreScript(previousFreezeUntilBlock);
+    const signature = (Transaction as any).sighash.sign(spendTransaction, this.privateKey, 0x1, 0, previousFreezeScript);

```

Comment:
Consider declaring const variable for `0x1` and `0` before passing to give indication of what they are.